src/SourceGenerators/Uno.UI.SourceGenerators/Helpers/HashBuilder.cs

```diff
@@ -28,25 +45,24 @@ public static string BuildIDFromSymbol(ITypeSymbol typeSymbol)
 		/// <param name="input">The string to hash</param>
 		/// <param name="algoritm">The algorithm to use</param>
 		/// <returns>A hex-formatted string of the hash</returns>
-		public static string Build(string input, HashAlgorithm algoritm = null)
+		public static string Build(string input)
 		{
-			using (algoritm ??= SHA256.Create())
+			using (var algorithm = SHA256.Create())
 			{
-				// Convert the input string to a byte array and compute the hash.
-				var data = algoritm.ComputeHash(Encoding.UTF8.GetBytes(input));
+				var data = algorithm.ComputeHash(Encoding.UTF8.GetBytes(input));
 
-				// Create a new Stringbuilder to collect the bytes
-				// and create a string.
-				var builder = new StringBuilder();
+				var min = Math.Min(data.Length, 16);
 
-				// Use the first 16 bytes of the hash
-				for (int i = 0; i < Math.Min(data.Length, 16); i++)
+				Span<char> span = stackalloc char[32];
+				for (int i = 0; i < min; i++)
 				{
-					builder.Append(data[i].ToString("x2", CultureInfo.InvariantCulture));
+					var s = _formattedBytes[data[i]];
+					span[i * 2] = s[0];
+					span[i * 2 + 1] = s[1];
 				}
 
-				// Return the hexadecimal string.
-				return builder.ToString();
+
+				return span.Slice(0, min).ToString();
```

Comment:
This bug was caught by the test `When_Grid_Uses_Both_Syntaxes` 🎉 

Looks like https://github.com/unoplatform/uno/pull/9801 was useful :) :)

```suggestion
				return span.Slice(0, min * 2).ToString();
```